### DISCUSSION TITLE
Initial setup and Install Solax X1 with Pocket LAN v3.0

### DISCUSSION THREAD

**Discussion by AlanKWatson68** [wills106/homeassistant-solax-modbus#506] [General]
Hi All

   like most people im new to this so forgive any stupid questions.....

I have Solax setup via the Solax Cloud API and generally all OK.
I looks like I should be able to get this information direct from the inveter and Modbus components as well.

So a few quick questions before I screw my HA install up getting it wrong.....

1 - I cant see this in the HACS list is it now only installable via the zip file?
2 - Ideally I want to control the charging functions of the inverter to get HA to decide when to charge and discharge the battery to the grid if I want to. I can do this manually with the Solax Web and App so hoping I should be able to script this?? any pointers would be useful.

Thanks for any help

**Comment by wills106** [comment]
You should be able to find it in HACS.
The PocketLAN doesn't provide a Modbus connection as far as I am aware of.

**Reply by AlanKWatson68** [reply]
Hi wills106 :)

    I cant find it listed in HACS and im told the pocket LAN should work by Solax....

I'll try the zip file version.

Have you used this for changing the charging modes of your inverter?

**Reply by wills106** [reply]
In HACS have you gone to Integrations and Searched at the top?

![image](https://github.com/wills106/homeassistant-solax-modbus/assets/18155231/3fcf6ca4-ad34-41c5-8bd5-5e3ca7a652c7)

Yes, aslong as the Inverter is awake you can change things such as Charger Use Mode and many other things.

**Reply by AlanKWatson68** [reply]
Got it :) cheers

**Reply by AlanKWatson68** [reply]
Im obviously a real newbie with this so forgive me..... Im happy to code but im sure im missing something
Inverter is a Solax X1-AC 3.6 and the setup originally was a "Social Energy" install so had modbus originally setup but that company is now no longer running so trying to use the hardware to control the inverter via HA 
As it was already installed there is a USB-RS485 connector installed and now setup on the Pi4 and can be seen in HA.
Once the correct baud rate is set 155200 as set by the inverter I can see some information but think im must be doing something wrong as only see :

Battery Awaken
Unlock Inverter
Unlock Inverter - Advanced

On the same modbus there are already 3 Eastron SDM220-Modbus meters all with different ID's.

Any suggestions?
I was hoping to see the inverter controls to automate the force charging but im sure im missing something.

**Reply by AlanKWatson68** [reply]
Sorted.....
Looks like it was conflicting with the Solax API calls.
Removed this and the Solax API calls - restarted and then re installed the Solax Modbus and hey presto :)

Now need to figure out some logic :)

**Comment by AlanKWatson68** [comment]
@wills106 just want to say thanks for this project its awesome💯 

Just working through the logic and cant seem to "charge now" type function, have you managed to get something to work?
Im an Octopus Mini user and have Power-Ups so ideally trying to make this slicker.... Octopus are going to add the feature in at some stage im told so want to see if I can automate this, if not its much easier than using the Solax App

**Reply by wills106** [reply]
You can either set times with:
select.solax_allow_grid_charge
select.solax_charger_use_mode
select.solax_charger_start_time_1 & select.solax_charger_end_time_1

Or if you want more dynamic control [look at this](https://github.com/wills106/homeassistant-solax-modbus/wiki/SolaX-Gen2-&-Gen3-Modes-of-operation#grid-export-gen-3-only)
The wiki is a bit out of date, but you can set it +6000 to -6000. Positive = Charge (Import), Negative = Discharge (Export)

**Reply by AlanKWatson68** [reply]
Perfect I will have a play
